From 0a8ac320244b16f2699a8bcf5973ec8540296b98 Mon Sep 17 00:00:00 2001
From: Steve Best <sbest@redhat.com>
Date: Mon, 10 Nov 2014 20:15:07 -0500
Subject: [powerpc] pseries: Quieten ibm, pcie-link-speed-stats warning

Message-id: <20141110201507.45973.12679.sendpatchset@ibm-p7r2-01.lab.bos.redhat.com>
Patchwork-id: 99817
O-Subject: [PATCH RHEL7.1 BZ1162287] powerpc/pseries: Quieten ibm, pcie-link-speed-stats warning
Bugzilla: 1162287
RH-Acked-by: Jarod Wilson <jarod@redhat.com>

RHBZ#:
------
https://bugzilla.redhat.com/show_bug.cgi?id=1162287

Description:
------------
The ibm,pcie-link-speed-stats isn't mandatory, so we shouldn't print
a high priority error message when missing. One example where we see
this is QEMU.

Reduce it to pr_debug.

Signed-off-by: Anton Blanchard <anton@samba.org>
Signed-off-by: Michael Ellerman <mpe@ellerman.id.au>

Brew:
-----
BE: http://brewweb.devel.redhat.com/brew/taskinfo?taskID=8220265
LE: http://brewweb.devel.redhat.com/brew/taskinfo?taskID=8220273

Upstream:
---------
http://git.kernel.org/cgit/linux/kernel/git/mpe/linux.git/commit/?h=next&id=7aa189c8f57f2141b8655c2a13c7486d0844d490

Test Status:
------------
Booted RHEL 7.1 LE guest on IBM's Power KVM (ibm-p8-kvm-03-guest-04) and the msg is no longer displayed during boot.

before the patch msg is displayed

[    0.069070] PCI: Probing PCI hardware
[    0.069300] no ibm,pcie-link-speed-stats property
[    0.069471] PCI host bridge to bus 0000:00
[    0.069584] pci_bus 0000:00: root bus resource [io  0x10000-0x1ffff] (bus add

after the patch msg is not displayed
[    0.052277] PCI: Probing PCI hardware
[    0.052413] PCI host bridge to bus 0000:00
[    0.052527] pci_bus 0000:00: root bus resource [io  0x10000-0x1ffff] (bus add

---------------------------------------------------------------
Steve Best

Proposed Patch:
---------------

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/platforms/pseries/pci.c b/arch/powerpc/platforms/pseries/pci.c
index c413ec1..8b766e4 100644
--- a/arch/powerpc/platforms/pseries/pci.c
+++ b/arch/powerpc/platforms/pseries/pci.c
@@ -133,7 +133,7 @@ int pseries_root_bridge_prepare(struct pci_host_bridge *bridge)
  of_node_put(pdn);
 
  if (rc) {
-  pr_err("no ibm,pcie-link-speed-stats property\n");
+  pr_debug("no ibm,pcie-link-speed-stats property\n");
   return 0;
  }
 
-- 
1.7.1